src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/AbstractConditionMapper.java

```diff
@@ -101,26 +139,37 @@ private void validateUpdatedFieldIsNotInPrimaryKey(String updatedField, String p
         }
     }
 
-    private String getUpdateSetClause(RequestWrapper request, String field, AttributeValue value) {
+    protected void validateUpdatedFieldIsNotInIndexKey(String updatedField, DynamoSecondaryIndex index) {
+        PrimaryKey primaryKey = index.getPrimaryKey();
+        if (updatedField.equals(primaryKey.getHashKey()) || (primaryKey.getRangeKey().isPresent()
+            && updatedField.equals(primaryKey.getRangeKey().get()))) {
+            throw new AmazonServiceException(
+                String.format("Cannot update attribute %s. This secondary index is part of the index key",
+                    updatedField));
+        }
+    }
+
+    private String getUpdateClause(RequestWrapper request, String field, AttributeValue value, String delimiter) {
         String fieldPlaceholder = MappingUtils.getNextFieldPlaceholder(request);
         request.putExpressionAttributeName(fieldPlaceholder, field);
         String valuePlaceholder = MappingUtils.getNextValuePlaceholder(request);
         request.putExpressionAttributeValue(valuePlaceholder, value);
-        return fieldPlaceholder + " = " + valuePlaceholder;
+        return fieldPlaceholder + delimiter + valuePlaceholder;
     }
 
     @VisibleForTesting
-    static Map<String, AttributeValue> parseUpdateExpression(RequestWrapper request,
-                                                             String conditionExpression) {
+    static List<? extends Map<String, AttributeValue>> parseUpdateExpression(
```

Comment:
It'd be better if the return type is more self-explanatory in what it is, e.g., a class called UpdateActions, with two instances of `Map<String, AttributeValue>`, `setActions` and `addActions`. Otherwise, a comment about what the return value means would be useful.

Reply:
yeah I was thinking similar even though it would change the usage in other places